src/main/java/io/cloudsoft/dbaccess/client/AbstractDatabaseAccessClient.java

```diff
@@ -106,142 +146,191 @@ public String getJdbcUrl(String user, String pass) {
+                    line = line.substring(i + 1, line.length() - " */".length());
+                }
+                if (!Strings.isBlank(line)) {
+                    result.add(line);
+                }
             }
-        } catch (Exception e) {
-            Exceptions.propagateIfFatal(e);
+            s.close();
+            return result;
         }
-    }
-
-    public void close(Connection connection) {
-        try {
-            if (connection != null) {
-                connection.close();
+        
+        public void runStatements(String description, List<String> statements) {
+            if (statements==null || statements.isEmpty()) return;
+            try (Statement statement = connection.createStatement()) {
+                for (String s : statements) {
+                    for (String si: parseSqlScript(templated(s))) {
+                        LOG.debug("Executing: "+si);
+                        statement.execute(si);
+                    }
+                }
+            } catch (SQLException e) {
+                throw Exceptions.propagate("Error executing SQL for "+description, e);
+            }            
+        }
+        
+        public void createUser() {
+            runStatements("create user", getCreateUserStatements());
+        }
+    
+        private void grantPermissions() {
+            runStatements("grant permissions", getGrantPermissionsStatements());
+        }
+    
+        private void grantCustomAccessScript() {
+            if (Strings.isBlank(accessScript)) return;
+            runStatements("grant custom access script", MutableList.<String>builder().addAll(accessScript.split("\n")).build());
+        }
+        
+        private void deleteUser() {
+            runStatements("delete user", getDeleteUserStatements());
+        }
+        
+        protected List<String> getGrantPermissionsStatements() {
+            List<String> permissionStatements = Lists.newArrayList();
+            
+            if (AccessModes.CUSTOM == accessMode) return permissionStatements;
+            
+            // at a minimum should have SELECT permission
+            permissionStatements.addAll(getGrantSelectPermissionStatements());
+            
+            if (AccessModes.READ_ONLY == accessMode) {
+                if (permissions != null && !permissionStatements.isEmpty()) {
+                    LOG.warn("Read-only mode specified; ignoring non-empty requested permissions "+permissions);
+                }
+                return permissionStatements;
             }
-        } catch (Exception e) {
-            Exceptions.propagateIfFatal(e);
+            
+            if (permissions == null || permissions.isEmpty()) {
+                LOG.warn("Access mode "+accessMode+" specified but no permissions granted (beyond default SELECT)");
+                return permissionStatements;
+            }
+    
+            for (Permissions permission : permissions) {
+                switch (permission) {
```

Comment:
no - some classes override `getGrantSelectPermissionStatements` directly.  the `getGrantPermissionStatements(String)` is a generic method; I will rename it so that is clearer.